thetis/callback.py

```diff
@@ -417,8 +375,61 @@ def minmax():
             tracer_min = self.solver_obj.comm.allreduce(tracer_min, op=MPI.MIN)
             tracer_max = self.solver_obj.comm.allreduce(tracer_max, op=MPI.MAX)
             return tracer_min, tracer_max
-        super(TracerOvershootCallBack, self).__init__(minmax,
-                                                      solver_obj,
-                                                      outputdir=outputdir,
-                                                      export_to_hdf5=export_to_hdf5,
-                                                      append_to_log=append_to_log)
+        super(TracerOvershootCallBack, self).__init__(minmax, solver_obj, **kwargs)
+
+
+class DetectorsCallback(DiagnosticCallback):
+    """Callback that writes the specified fields interpolated in the specified detector locations"""
+    name = 'detectors'
```

Comment:
Will this work for multiple instances? Typically you do not want to extract all the variables in the same locations. E.g. in 3D you'll have different stations for temperature/salinity and elevation. If you define two instances will they try to write to the same file?

Reply:
I see, good point. I suppose I could have the user specify a `name` for each detector callback?

Reply:
Yes, that's probably the only way to do it. Should the name be compulsory so that users do not accidentally create two overlapping instances?